src/compiler/checker.ts

```diff
@@ -35405,7 +35405,7 @@ namespace ts {
                                         superCallStatement = statement;
                                         break;
                                     }
-                                    if (!isPrologueDirective(statement) && nodeImmediatelyReferencesSuperOrThis(statement)) {
+                                    if (nodeImmediatelyReferencesSuperOrThis(statement)) {

```

Comment:
This change isn't necessary for the PR; I just had only noticed the check is unnecessary _after_ #29374 was merged 😄